### PR TITLE
Add test for closing wasi preopen fd

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
As discussed in WASI, we should be able to close preopen fds

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors retry

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

File: lib/wasi/src/state/mod.rs [hunk 1]
```diff
@@ -941,6 +941,61 @@ impl WasiFs {
...
+                                        idx = Some(i);
+                                        break;
+                                    }
+                                }
+                                if let Some(i) = idx {
+                                    // only remove entry properly if this is the original preopen FD
+                                    // calling `path_open` can give you an fd to the same inode as a preopen fd
+                                    entries.remove(&key);
+                                    self.preopen_fds.remove(i);
+                                    // Maybe recursively closes fds if og preopen?
```
Comment (nlewycky) [inline lib/wasi/src/state/mod.rs]: "og"?

**Pull request merged**